tl/controllers/middleware.rst

```diff
@@ -129,121 +129,120 @@ scripts, that add middleware::
 
-    // Read the file data.
+    // Basahin ang datos ng file.
     $files[0]->getStream();
     $files[0]->getSize();
     $files[0]->getClientFileName();
 
-    // Move the file.
+    // Ilipat ang file.
     $files[0]->moveTo($targetPath);
 
-Requests contain a URI object, which contains methods for interacting with the
-requested URI::
+Ang mga hiling ay naglalaman ng URI object, na kung saan ay naglalaman ng mga pamamaraan para sa pag-ugnayan sa mga hiniling na URI::
 
-    // Get the URI
+    // Kunin ang URI
     $uri = $request->getUri();
 
-    // Read data out of the URI.
+    // Basahin ang datos na inilabas sa URI.
     $path = $uri->getPath();
     $query = $uri->getQuery();
     $host = $uri->getHost();
 
-Lastly, you can interact with a request's 'attributes'. CakePHP uses these
-attributes to carry framework specific request parameters. There are a few
-important attributes in any request handled by CakePHP:
+Panghuli, maaari kang mag-ugnay sa isang humihiling na 'attributes'. Ang CakePHP ay gumagamit nito
+na mga katangian upang dalhin ang framework sa tiyak na mga parameter ng kahilingan. Mayroong kunting
+importante na mga katangian sa anumang hiling na hinawakan ni CakePHP:
 
-* ``base`` contains the base directory for your application if there is one.
-* ``webroot`` contains the webroot directory for your application.
-* ``params`` contains the results of route matching once routing rules have been
-  processed.
-* ``session`` contains an instance of CakePHP's ``Session`` object. See
-  :ref:`accessing-session-object` for more information on how to use the session
-  object.
+* ``base`` ay naglalaman ng base na direktoryo para sa iyong aplikasyon kung meron mang isa.
+* ``webroot`` ay naglalaman ng webroot na direktoryo para sa iyong aplikasyon.
+* ``params`` ay naglalaman ng mga resulta sa ruta na tumutugma kapang ang mga patakaran ng pag-ruta ay
+  nai-proseso.
+* ``session`` ay naglalaman ng isang instance sa CakePHP ``Session`` na object. Tingnan
+  :ref:`accessing-session-object` para sa karagdagang impormasyon kung papaano gamitin ang sesyon
+  na object.
 
-Interacting with Responses
+Pag-uugnay ng mga Tugon
 --------------------------
 
-The methods available to create a server response are the same as those
-available when interacting with :ref:`httpclient-response-objects`. While the
-interface is the same the usage scenarios are different.
+Ang pamamaraan ay magagamit upang lumikha ng server na tugon ay pareho sa mga
+magagamit na iyon kapag nag-uugnay sa :ref:`httpclient-response-objects`. Habang ang
+interface ay pareho lamang ang gamit ng mga sitwasyon ay magkaiba.
 
-When modifying the response, it is important to remember that responses are
-**immutable**. You must always remember to store the results of any setter
-method. For example::
+Kapag nagbago sa tugon, ito ay importante na tandaan na ang mga tugon ay
+**immutable**. Dapat mong laging tandaan na mag-imbak ng mga resulta sa anumang mga setter
```

Comment:
- change immutable to hindi nababago
- Dapat mo laging

Reply:
done change